core/scripts/cli/sponsor/listMarkets.js

```diff
@@ -0,0 +1,27 @@
+const inquirer = require("inquirer");
+const style = require("../textStyle");
+
+const listMarkets = async (web3, artifacts) => {
+  style.spinnerReadingContracts.start();
+  const ExpiringMultiParty = artifacts.require("ExpiringMultiParty");
+  const Governor = artifacts.require("Governor");
+  const Registry = artifacts.require("Registry");
+
+  const governorAddress = (await Governor.deployed()).address;
+
+  const registry = await Registry.deployed();
+  const contractAddresses = await registry.getAllRegisteredContracts();
```

Comment:
nit: do you think it's better to use this event https://github.com/UMAprotocol/protocol/blob/b066d69bb3e3cb7e7e2d35eb73a5216e70dfbf2e/core/contracts/financial-templates/implementation/ExpiringMultiPartyCreator.sol#L79 rather than the Registry to avoid picking up contracts that aren't EMPs? Right now, we only have to special case the governor so this isn't a huge deal either way.

Reply:
It just seems to be *much* slower than reading the Registry, at least locally on Ganache for me. Unlike the liquidation bot, this CLI is meant to be an interactive tool, so maybe there's an advantage to the faster approach?

Reply:
Seems reasonable to me. My guess is that the speed difference would be even larger on mainnet.